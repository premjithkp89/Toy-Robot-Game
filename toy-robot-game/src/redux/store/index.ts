import { configureStore } from "@reduxjs/toolkit";
import placeRobotReducer from "../reducers/inputReducer";

export const store = configureStore({
  reducer: {
    placeRobot: placeRobotReducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
