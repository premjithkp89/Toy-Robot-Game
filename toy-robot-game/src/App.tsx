import {RobotApp} from './pages/robot/RobotApp';
import './App.css';
import { ThemeProvider } from '@material-ui/core'
import { createTheme } from '@material-ui/core/styles';
import { store } from './redux/store';
import { Provider } from 'react-redux';

const theme = createTheme({
  overrides: {
      // MuiTableCell: {
      //     root: {  //This can be referred from Material UI API documentation.
      //         padding: '0px 0px',
      //         backgroundColor: 'gray',

      //     },
      // },

      // MuiTableRow:{
      //   root: {  //This can be referred from Material UI API documentation.
      //     padding: '0px 0px',
      //     backgroundColor: 'gray',

      // },

      // },



  },
});

function App() {
  return (

    <div className="App">
      <header className="App-header">
      <ThemeProvider theme={theme}>
      <Provider store={store}>
      <RobotApp/>
      </Provider>
      </ThemeProvider>
      </header>
    </div>
  );
}

export default App;



