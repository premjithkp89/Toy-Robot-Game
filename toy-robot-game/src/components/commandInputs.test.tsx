
import "@testing-library/jest-dom";

import configureStore from 'redux-mock-store';
import {store} from '../redux/store'
import {
  render,
  screen,
  fireEvent,
  waitFor,
} from '@testing-library/react';
import { Provider } from 'react-redux';

import App from '../App';


const initialState = { xCordinate: 0,
  yCordinate: 0,
  row: -1,
  column: -1,
  wallRow: -1,
  wallColumn: -1,
  direction: "",
  angle: 0,
  showrobot: false,
  showReport:false,
  wallMap: { "": false },};
  const mockStore = configureStore();

  const stores = mockStore(initialState);

    describe("<CommandInputs>", () => {
     it('should place a robot when place button is clicked',  async ()=> {

        render(
          <Provider store={stores}>
              <App />
          </Provider>
      );
        const placeRobot =screen.queryByTestId('place-robot')
        placeRobot && fireEvent.click(placeRobot)

        await waitFor(() =>  {
          expect(screen.getByTestId("robot")).toBeInTheDocument();

        });
    });

    it('should  move the robot when move button clicked',  async ()=> {
      render(
        <Provider store={store}>
            <App />
        </Provider>
    );

    const state = store.getState().placeRobot;
    const initialXCordinate = state.xCordinate;
    const initialYCordinate = state.yCordinate;

      const moveRobot =screen.queryByTestId('move-robot')
      moveRobot && fireEvent.click(moveRobot)

      await waitFor(() =>  {
        const finalXCordinate = state.xCordinate;
        const finalYCordinate = state.yCordinate;

        expect( initialXCordinate !== finalXCordinate || initialYCordinate !== finalYCordinate).toBeFalsy();

      });
  });
  });


