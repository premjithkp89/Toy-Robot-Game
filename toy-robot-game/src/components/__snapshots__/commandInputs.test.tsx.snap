// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the app 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <div
        style="display: flex; flex-direction: row; justify-content: space-between;"
      >
        <div
          class="MuiBox-root MuiBox-root-2"
        >
          <div
            class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
          >
            <table
              aria-label="simple table"
              class="MuiTable-root makeStyles-table-1"
            >
              <tbody
                class="MuiTableBody-root"
              >
                <tr
                  class="MuiTableRow-root"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body"
                    style="padding: 0px;"
                  />
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-3"
                      id="5-1"
                      style="background-image: url(cell.png); background-size: contain; width: 80px; padding: 1px 1px; height: 80px;"
                    />
                  </td>
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-4"
                      id="5-2"
                      style="background-image: url(cell.png); background-size: contain; width: 80px; padding: 1px 1px; height: 80px;"
                    />
                  </td>
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-5"
                      id="5-3"
                      style="background-image: url(cell.png); background-size: contain; width: 80px; padding: 1px 1px; height: 80px;"
                    />
                  </td>
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-6"
                      id="5-4"
                      style="background-image: url(cell.png); background-size: contain; width: 80px; padding: 1px 1px; height: 80px;"
                    />
                  </td>
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-7"
                      id="5-5"
                      style="background-image: url(cell.png); background-size: contain; width: 80px; padding: 1px 1px; height: 80px;"
                    />
                  </td>
                </tr>
                <tr
                  class="MuiTableRow-root"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body"
                    style="padding: 0px;"
                  />
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-8"
                      id="4-1"
                      style="background-image: url(cell.png); background-size: contain; width: 80px; padding: 1px 1px; height: 80px;"
                    />
                  </td>
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-9"
                      id="4-2"
                      style="background-image: url(cell.png); background-size: contain; width: 80px; padding: 1px 1px; height: 80px;"
                    />
                  </td>
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-10"
                      id="4-3"
                      style="background-image: url(cell.png); background-size: contain; width: 80px; padding: 1px 1px; height: 80px;"
                    />
                  </td>
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-11"
                      id="4-4"
                      style="background-image: url(cell.png); background-size: contain; width: 80px; padding: 1px 1px; height: 80px;"
                    />
                  </td>
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-12"
                      id="4-5"
                      style="background-image: url(cell.png); background-size: contain; width: 80px; padding: 1px 1px; height: 80px;"
                    />
                  </td>
                </tr>
                <tr
                  class="MuiTableRow-root"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body"
                    style="padding: 0px;"
                  />
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-13"
                      id="3-1"
                      style="background-image: url(cell.png); background-size: contain; width: 80px; padding: 1px 1px; height: 80px;"
                    />
                  </td>
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-14"
                      id="3-2"
                      style="background-image: url(cell.png); background-size: contain; width: 80px; padding: 1px 1px; height: 80px;"
                    />
                  </td>
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-15"
                      id="3-3"
                      style="background-image: url(cell.png); background-size: contain; width: 80px; padding: 1px 1px; height: 80px;"
                    />
                  </td>
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-16"
                      id="3-4"
                      style="background-image: url(cell.png); background-size: contain; width: 80px; padding: 1px 1px; height: 80px;"
                    />
                  </td>
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-17"
                      id="3-5"
                      style="background-image: url(cell.png); background-size: contain; width: 80px; padding: 1px 1px; height: 80px;"
                    />
                  </td>
                </tr>
                <tr
                  class="MuiTableRow-root"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body"
                    style="padding: 0px;"
                  />
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-18"
                      id="2-1"
                      style="background-image: url(cell.png); background-size: contain; width: 80px; padding: 1px 1px; height: 80px;"
                    />
                  </td>
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-19"
                      id="2-2"
                      style="background-image: url(cell.png); background-size: contain; width: 80px; padding: 1px 1px; height: 80px;"
                    />
                  </td>
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-20"
                      id="2-3"
                      style="background-image: url(cell.png); background-size: contain; width: 80px; padding: 1px 1px; height: 80px;"
                    />
                  </td>
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-21"
                      id="2-4"
                      style="background-image: url(cell.png); background-size: contain; width: 80px; padding: 1px 1px; height: 80px;"
                    />
                  </td>
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-22"
                      id="2-5"
                      style="background-image: url(cell.png); background-size: contain; width: 80px; padding: 1px 1px; height: 80px;"
                    />
                  </td>
                </tr>
                <tr
                  class="MuiTableRow-root"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body"
                    style="padding: 0px;"
                  />
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-23"
                      id="1-1"
                      style="background-image: url(cell.png); background-size: contain; width: 80px; padding: 1px 1px; height: 80px;"
                    />
                  </td>
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-24"
                      id="1-2"
                      style="background-image: url(cell.png); background-size: contain; width: 80px; padding: 1px 1px; height: 80px;"
                    />
                  </td>
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-25"
                      id="1-3"
                      style="background-image: url(cell.png); background-size: contain; width: 80px; padding: 1px 1px; height: 80px;"
                    />
                  </td>
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-26"
                      id="1-4"
                      style="background-image: url(cell.png); background-size: contain; width: 80px; padding: 1px 1px; height: 80px;"
                    />
                  </td>
                  <td
                    style="padding: 0px;"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-27"
                      id="1-5"
                      style="background-image: url(cell.png); background-size: contain; width: 80px; padding: 1px 1px; height: 80px;"
                    />
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          class="makeStyles-container-28"
        >
          <div
            class="makeStyles-placeRobotContainer-29"
          >
            <div>
              <div
                class="MuiFormControl-root"
                style="margin-right: 10px;"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
                  data-shrink="true"
                >
                  Rows
                  <span
                    aria-hidden="true"
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                  >
                     *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                >
                  <div
                    aria-haspopup="listbox"
                    aria-labelledby="mui-component-select-rows"
                    class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                    id="mui-component-select-rows"
                    role="button"
                    tabindex="0"
                  >
                    <span>
                      ​
                    </span>
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput"
                    name="rows"
                    required=""
                    tabindex="-1"
                    value="-1"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-31 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-33 PrivateNotchedOutline-legendNotched-34"
                    >
                      <span>
                        Rows
                      </span>
                    </legend>
                  </fieldset>
                </div>
                <p
                  class="MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-filled Mui-required"
                >
                  Select row
                </p>
              </div>
              <div
                class="MuiFormControl-root"
                style="margin-right: 10px;"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
                  data-shrink="true"
                >
                  Column
                  <span
                    aria-hidden="true"
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                  >
                     *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                >
                  <div
                    aria-haspopup="listbox"
                    aria-labelledby="mui-component-select-cols"
                    class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                    id="mui-component-select-cols"
                    role="button"
                    tabindex="0"
                  >
                    <span>
                      ​
                    </span>
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput"
                    name="cols"
                    required=""
                    tabindex="-1"
                    value="-1"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-31 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-33 PrivateNotchedOutline-legendNotched-34"
                    >
                      <span>
                        Column
                      </span>
                    </legend>
                  </fieldset>
                </div>
                <p
                  class="MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-filled Mui-required"
                >
                  Select column
                </p>
              </div>
              <div
                class="MuiFormControl-root"
                style="margin-right: 10px;"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-required Mui-required"
                  data-shrink="true"
                >
                  Direction
                  <span
                    aria-hidden="true"
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                  >
                     *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                >
                  <div
                    aria-haspopup="listbox"
                    aria-labelledby="mui-component-select-direction"
                    class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                    id="mui-component-select-direction"
                    role="button"
                    tabindex="0"
                  >
                    <span>
                      ​
                    </span>
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput"
                    name="direction"
                    required=""
                    tabindex="-1"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-31 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-33"
                    >
                      <span>
                        Direction
                      </span>
                    </legend>
                  </fieldset>
                </div>
                <p
                  class="MuiFormHelperText-root MuiFormHelperText-contained Mui-required"
                >
                  Select direction
                </p>
              </div>
            </div>
            <div
              style="height: 0px;"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary Mui-disabled Mui-disabled"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  PLACE ROBOT
                </span>
              </button>
            </div>
          </div>
          <div
            class="makeStyles-commandButtonContainer-30"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                LEFT
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                RIGHT
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                MOVE
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="makeStyles-placeRobotContainer-29"
          >
            <div
              class="MuiFormControl-root"
              style="margin-right: 10px;"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
                data-shrink="true"
              >
                Row
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <div
                  aria-haspopup="listbox"
                  aria-labelledby="mui-component-select-wall-rows"
                  class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                  id="mui-component-select-wall-rows"
                  role="button"
                  tabindex="0"
                >
                  <span>
                    ​
                  </span>
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput"
                  name="wall-rows"
                  required=""
                  tabindex="-1"
                  value="-1"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-31 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-33 PrivateNotchedOutline-legendNotched-34"
                  >
                    <span>
                      Row
                    </span>
                  </legend>
                </fieldset>
              </div>
              <p
                class="MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-filled Mui-required"
              >
                Select row
              </p>
            </div>
            <div
              class="MuiFormControl-root"
              style="margin-right: 10px;"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
                data-shrink="true"
              >
                Column
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <div
                  aria-haspopup="listbox"
                  aria-labelledby="mui-component-select-wall-cols"
                  class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                  id="mui-component-select-wall-cols"
                  role="button"
                  tabindex="0"
                >
                  <span>
                    ​
                  </span>
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput"
                  name="wall-cols"
                  required=""
                  tabindex="-1"
                  value="-1"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-31 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-33 PrivateNotchedOutline-legendNotched-34"
                  >
                    <span>
                      Column
                    </span>
                  </legend>
                </fieldset>
              </div>
              <p
                class="MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-filled Mui-required"
              >
                Select column
              </p>
            </div>
            <div
              style="height: 0px;"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary Mui-disabled Mui-disabled"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  PLACE WALL
                </span>
              </button>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              REPORT
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <div>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              ROW:-1 
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              COL:-1
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              FACING:
            </p>
          </div>
        </div>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
